python_ta/cfg/graph.py

```diff
@@ -245,3 +341,86 @@ def __init__(
         self.condition = condition
         self.source.successors.append(self)
         self.target.predecessors.append(self)
+        self.z3_constraints = {}
+
+
+class Z3Environment:
+    """Z3 Environment stores the Z3 variables and constraints in the current CFG path
+
+    variable_unassigned:
+        A dictionary mapping each variable in the current environment to a boolean indicating
+        whether it has been reassigned (False) or remains unassigned (True).
+
+    variable_type:
+        A dictionary mapping each variable in the current environment to its data type.
+
+    constraints:
+        A list of Z3 constraints in the current environment.
+    """
+
+    variable_unassigned: Dict[str, bool]
+    variable_type: Dict[str, str]
```

Comment:
Rename this to just `variables`. Make sure to update the type annotation and the class docstring.